lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -497,12 +498,25 @@ class TypeAnalysis : public dataflow::SparseDataFlowAnalysis<
 // operation (such as AtenMulScalarOp, AtenAddScalarOp etc)
 // If the data is floating-point, the `dtype` is inferred to be the
 // default dtype, see `torch.get_default_dtype`.
-static Type getDefaultDtypeForTorchScalar(Type type) {
+static Type getDefaultDtypeForTorchScalar(Operation *op, Type type) {
   MLIRContext *context = type.getContext();
   if (type.isa<Torch::FloatType>()) {
     // For now, use float32 which is the initial default dtype returned by
     // `torch.get_default_dtype`.
-    return Float32Type::get(context);
+    auto module = op->getParentOfType<ModuleOp>();
+    auto default_dtype =
+        module->getAttr("torch_mlir.default_dtype").cast<StringAttr>().str();
+
+    Type dtype;
+    if (!default_dtype.compare("torch.float16")) {
+      dtype = Float16Type::get(context);
+    } else if (!default_dtype.compare("torch.float64")) {
+      dtype = Float64Type::get(context);
+    } else {
+      dtype = Float32Type::get(context);
```

Comment:
nit: can you use an `else if ` for the `float32` case as well and have the `else` finish with an `llvm_unreachable`? This makes it very clear that every possible case is being handled appropriately